src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -25,6 +25,52 @@
 
 namespace ekat {
 
+namespace impl {
+
+template <bool Serialize, typename TeamMember, typename Lambda, typename ValueType>
```

Comment:
@bartgol I'm concerned about cluttering code with the Serialize parameter. Would EKAT support putting this last with a config-time default, something like `template <..., bool Serialize=EKAT_BFB_OPS>`?

Another option is SCREAM can have a wrapper to this wrapper that handles the Serialize parameter so the actual app doesn't have to provide any serialize argument to `parallel_reduce`.

Reply:
I wrote it as a comment somewhere, but it got lost with the push. Bottom line is that I'd like to not put bfb macros in ekat. TBC, it's just my preference. I just think of need for bfb as a "customer need" (to be addressed in the app), not a "provider service" (to appear in the library). Especially if the macro had a very limited number of places where used (like, in this case, one).

One limitation of adding the macro in EKAT is that it forces _all_ use cases to do the same thing. Although this might not be a big deal in practice, it is a limitation of sort, that can easily be avoided.

That said, as I wrote in my (lost) original comment, scream can easily define a wrapper function, with CPP logic to pick the instantiation.

Reply:
Another solution could be to add the `Serialize` template argument directly to the class, so that customers can do

```
template<typename ExecSpace>
#ifdef BFB
using ExecSpaceUtils = ekat::ExecSpaceUtils<ExecSpace,false>;
#else
using ExecSpaceUtils = ekat::ExecSpaceUtils<ExecSpace,false>;
#endif
```
so that the CPP logic is done once for all exec utils.

Reply:
Luca, that's fine. Tbc, I strongly disagree. EKAT should provide solutions to exactly what the common needs are, including handling BFB-testing needs. But I defer to yours on matters of opinion. Adding some type of wrapper in the app is fine.

Reply:
Personally, I'm on the fence about including BFB stuff in EKAT, because I loathe the methodology and prefer libraries with a light touch. But the need for BFB reflects the reality of our work. So I could go either way at the moment (stick it in now, or wait for folks to ask for EKAT support for it).

Reply:
I opened an issue, let's discuss.